core/engine/src/string.rs

```diff
@@ -0,0 +1,256 @@
+//! A Latin1 or UTF-16 encoded, reference counted, immutable string.
+//!
+//! This module contains the [`JsString`] type, the [`js_string`][crate::js_string] macro and the
+//! [`js_str`][crate::js_str] macro.
+//!
+//! The [`js_string`][crate::js_string] macro is used when you need to create a new [`JsString`],
+//! and the [`js_str`][crate::js_str] macro is used for const conversions of string literals to [`JsStr`].
+
+// Required per unsafe code standards to ensure every unsafe usage is properly documented.
+// - `unsafe_op_in_unsafe_fn` will be warn-by-default in edition 2024:
+//   https://github.com/rust-lang/rust/issues/71668#issuecomment-1189396860
+// - `undocumented_unsafe_blocks` and `missing_safety_doc` requires a `Safety:` section in the
+//   comment or doc of the unsafe block or function, respectively.
+#![deny(
+    unsafe_op_in_unsafe_fn,
+    clippy::undocumented_unsafe_blocks,
+    clippy::missing_safety_doc
+)]
+// Remove when/if https://github.com/rust-lang/rust/issues/95228 stabilizes.
+// Right now this allows us to use the stable polyfill from the `sptr` crate, which uses
+// the same names from the unstable functions of the `std::ptr` module.
+#![allow(unstable_name_collisions)]
```

Comment:
These docs and rules can probably be altered since the only thing in this file is the `js_string` macro right?